marklogic/datadog_checks/marklogic/api.py

```diff
@@ -0,0 +1,106 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from typing import Any, Dict
+
+from datadog_checks.base.utils.http import RequestsWrapper
+
+
+class MarkLogicApi(object):
+    def __init__(self, http, api_url):
+        # type: (RequestsWrapper, str) -> None
+        self._http = http
+        self._base_url = api_url + '/manage/v2'
+
+    def http_get(self, route="", params=None):
+        # type: (str, Dict[str, str]) -> Dict[str, Any]
+        if params is None:
+            params = {}
+        params['format'] = 'json'  # Always in json
+
+        url = self._base_url + route
+        resp = self._http.get(url, params=params)
+        resp.raise_for_status()
+        return resp.json()
+
+    def get_status_data(self, resource=None, name=None, group=None):
+        # type: (str, str, str) -> Dict[str, Any]
+        """
+        Example url:
+            - http://localhost:8002/manage/v2/hosts?view=status
+            - http://localhost:8002/manage/v2/hosts?view=status&format=json (cluster level)
+            - http://localhost:8002/manage/v2/forests/Security?view=status&format=json
+            - http://localhost:8002/manage/v2/databases/Extensions?view=status&format=json
+            - http://localhost:8002/manage/v2/hosts/2871b05b4bdc?view=status&format=json
+        """
+        params = {'view': 'status'}
+        route = ""
+        if resource:
```

Comment:
Yes, [here](https://github.com/DataDog/integrations-core/pull/6524/files#diff-d51841768409f70532ea8dc0ae3e5a92R88)

However this method should be cleaned because we use [uri](https://github.com/DataDog/integrations-core/pull/6524/files#diff-d51841768409f70532ea8dc0ae3e5a92R142) to get the status metrics.